README.md

```diff
@@ -190,30 +190,42 @@ easy to fork and contribute any changes back upstream.
 
 2. **Define environment variable `PYENV_ROOT`** to point to the path where
    pyenv repo is cloned and add `$PYENV_ROOT/bin` to your `$PATH` for access
-   to the `pyenv` command-line utility.
-
+   to the `pyenv` command-line utility. For example, if you would like to put
+   environment variables in `~/.bash_profile`:
     ```sh
     $ echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bash_profile
     $ echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bash_profile
     ```
+    
+    - **Linux note**: Your desktop environment may not source `~/.bash_profile`.
+      If this happens, you may want to use `~/.profile` instead, since it's usually
+      sourced by the display manager.
     - **Zsh note**: Modify your `~/.zshenv` file instead of `~/.bash_profile`.
     - **Ubuntu and Fedora note**: Modify your `~/.bashrc` file instead of `~/.bash_profile`.
     - **Proxy note**: If you use a proxy, export `http_proxy` and `HTTPS_PROXY` too.
 
 3. **Add `pyenv init` to your shell** to enable shims and autocompletion.
    Please make sure `eval "$(pyenv init -)"` is placed toward the end of the shell
-   configuration file since it manipulates `PATH` during the initialization.
+   configuration file since it manipulates `PATH` during the initialization. For example:
     ```sh
     $ echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> ~/.bash_profile
     ```
+    
+    If you put this in the run control file (like `~/.bashrc`), this will work fine; otherwise (like if you
+    use `~/.bash_profile` or `~/.profile` instead, and it is not sourced in `~/.bashrc`), you need to put
+    `export -f pyenv` after this to access the function from a sub-shell.
```

Comment:
Should/could `export -f pyenv` instead somehow be part of `pyenv init -`? That would avoid adding any extra steps for users to do.

(FWIW, I found this tip via #1347. I was having the same issue with sub-shells and your suggestion there fixed things for me. Thanks!)